app/javascript/controllers/pagination_controller.js

```diff
@@ -0,0 +1,69 @@
+import { Controller } from "@hotwired/stimulus"
+import { get } from '@rails/request.js';
+
+export default class extends Controller {
+    static targets = ['lastPage', 'loadMoreContainer', 'skeleton']
+
+    static values = {
+        url: String,
+        page: Number,
+    };
+
+    initialize() {
+        this.bodyScrollListener = this.bodyScrollListener.bind(this);
```

Comment:
I strongly prefer using `IntersectionObserver` over a scroll listener. @excide3 has a great [GoRails example with some sample code](https://github.com/gorails-screencasts/infinite-scroll-stimulus-js/blob/master/app/javascript/controllers/infinite_scroll_controller.js) showing how to do this.

Also, we should be able to use the next page `rel` links that `pagy` autogenerates (and we hide) to tell if there are more pages. We shouldn't need a custom HTML tag.

Reply:
Alright, so i've ended up using the controller provided by Chris. The first time the pagy links are intersected it works fine, but after that it does not work. Any ideas why?


https://user-images.githubusercontent.com/38360603/140611916-be5a75c8-5ab1-4c42-b9c4-828c0b76e993.mov

Reply:
Can you set a `debugger` breakpoint or log out in the `loadMore()` method to make sure: 

1. It get's called
1. `next_page` isn't `null`
2. `next_page.href` isn't `null`
3. Add an error callback and see if the XHR is failing

Reply:
I've found what was causing the issue. When i was replacing the pagy links 
```erb
<%= turbo_stream.replace "pagination" do %>
  <div id="pagination" data-pagination-target="pagination">
    <%== pagy_nav(@pagy) %>
  </div>
<% end %>
```

the observer was getting disconnected. So, i'm listening to connection event and simply reattaching the observer.

```js
    paginationTargetConnected() {
        this.intersectionObserver.observe(this.paginationTarget)
    }
```